src/test/java/io/stargate/sgv2/jsonapi/service/cqldriver/CqlSessionCacheTests.java

```diff
@@ -0,0 +1,254 @@
+    // Throwable
+    Throwable t = catchThrowable(cqlSessionCacheForTest::getSession);
+    assertThat(t).isNotNull().isInstanceOf(UnauthorizedException.class).hasMessage("Unauthorized");
+    // metrics test
+    Gauge cacheSizeMetric =
+        meterRegistry.find("cache.size").tag("cache", "cql_sessions_cache").gauge();
+    FunctionCounter cachePutMetric =
+        meterRegistry.find("cache.puts").tag("cache", "cql_sessions_cache").functionCounter();
+    assertThat(cacheSizeMetric).isNotNull();
+    assertThat(cacheSizeMetric.value()).isEqualTo(0);
+    assertThat(cachePutMetric).isNotNull();
+    assertThat(cachePutMetric.count()).isEqualTo(0);
+  }
+
+  @Test
+  public void testOSSCxCQLSessionCacheMultiTenant()
+      throws NoSuchFieldException, IllegalAccessException {
+    CQLSessionCache cqlSessionCacheForTest = new CQLSessionCache(operationsConfig, meterRegistry);
+    // set operation config
+    Field operationsConfigField =
+        cqlSessionCacheForTest.getClass().getDeclaredField("operationsConfig");
+    operationsConfigField.setAccessible(true);
+    operationsConfigField.set(cqlSessionCacheForTest, operationsConfig);
+    List<String> tenantIds = new ArrayList<>();
+    tenantIds.add("tenant1");
+    tenantIds.add("tenant2");
+    tenantIds.add("tenant3");
+    tenantIds.add("tenant4");
+    tenantIds.add("tenant5");
+    for (String tenantId : tenantIds) {
+      // set request info
+      StargateRequestInfo stargateRequestInfo = mock(StargateRequestInfo.class);
+      when(stargateRequestInfo.getTenantId()).thenReturn(Optional.of(tenantId));
+      when(stargateRequestInfo.getCassandraToken())
+          .thenReturn(operationsConfig.databaseConfig().fixedToken());
+      Field stargateRequestInfoField =
+          cqlSessionCacheForTest.getClass().getDeclaredField("stargateRequestInfo");
+      stargateRequestInfoField.setAccessible(true);
+      stargateRequestInfoField.set(cqlSessionCacheForTest, stargateRequestInfo);
+      CqlSession cqlSession = cqlSessionCacheForTest.getSession();
+      sessionsCreatedInTests.add(cqlSession);
+      assertThat(
+              ((DefaultDriverContext) cqlSession.getContext())
+                  .getStartupOptions()
+                  .get(TENANT_ID_PROPERTY_KEY))
+          .isEqualTo(tenantId);
+    }
+    // metrics test
+    Gauge cacheSizeMetric =
+        meterRegistry.find("cache.size").tag("cache", "cql_sessions_cache").gauge();
+    assertThat(cacheSizeMetric).isNotNull();
+    assertThat(cacheSizeMetric.value()).isEqualTo(tenantIds.size());
+    FunctionCounter cachePutMetric =
+        meterRegistry.find("cache.puts").tag("cache", "cql_sessions_cache").functionCounter();
+    assertThat(cachePutMetric).isNotNull();
+    assertThat(cachePutMetric.count()).isEqualTo(tenantIds.size());
+    FunctionCounter cacheLoadMetric =
+        meterRegistry
+            .find("cache.load")
+            .tag("cache", "cql_sessions_cache")
+            .tag("result", "success")
```

Comment:
Do you think we may have a cache load failure?
Not familiar with Metrics, I assume we can pass the test even without line 198.

Reply:
Yes, there might be failures in case if a session can not be established with the backend, so checking for `success` tag also here.